build-tests/api-extractor-scenarios/tsconfig.json

```diff
@@ -10,7 +10,14 @@
     "strictNullChecks": true,
     "types": ["node", "jest"],
     "lib": ["es5", "scripthost", "es2015.collection", "es2015.promise", "es2015.iterable", "dom"],
-    "outDir": "lib"
+    "outDir": "lib",
+
+    // These properties are only used by the pathMappings/ test scenario.
+    "baseUrl": ".",
```

Comment:
Does your approach work with a baseUrl that is something other than `"."`?  If not, should we report a warning for an unsupported configuration?

Reply:
Yes, my approach is compatible with any `baseUrl` value. It doesn't process the `baseUrl` and doesn't assume a particular value.

Reply:
> The TypeScript docs [give an example](https://www.typescriptlang.org/tsconfig#baseUrl) where `"baseUrl": "."` is used without `"paths"`. Do we support that? If not, should we issue a warning in that case?

Let me check... I'm not sure about this yet.

Reply:
Thanks for flagging this, it exposed an issue in this PR that I hadn't considered. Suppose a project has the following tsconfig:

```
{
  "baseUrl": "<anything>",
}
```

And writes an import like this:

```
import {jquery} from 'jquery';
```

Does this a local file from within the tsconfig directory called `./jquery.ts` or from an external NPM package called `jquery`? In practice, the TS Compiler will look for both - first it will try to look for a local file, then it will try to resolve any `node_modules` matches. So the PR as-is doesn't quite work, because any non-relative import could potentially be resolving to a local file if `baseUrl` is used.

So I took a different approach entirely. I refactored `isExternalModulePath` to instead use the `TypeScriptInternals.getResolvedModule` helper in order to determine whether a module is external (still taking into consideration any bundled packages). There's a helpful flag set on the resolved module called `isExternalLibraryImport` that AFAICT is true IFF the resolved path contains the string `node_modules` somewhere (not exactly sure what the logic is, though). In practice it seems to work really well.

This means that relative imports into `node_modules` (e.g. `../../node_modules/jquery/...` as well as path mappings that resolve to `node_modules` are both considered to be external modules, unless of course bundled. Altogether, I think this is a solid improvement over the previous logic.

Reply:
Nice! 😎👍